src/core/pathfinding/PathFinding.ts

```diff
@@ -2,9 +2,56 @@ import { consolex } from "../Consolex";
 import { Game } from "../game/Game";
 import { GameMap, TileRef } from "../game/GameMap";
 import { PseudoRandom } from "../PseudoRandom";
+import { DistanceBasedBezierCurve } from "../utilities/Line";
 import { AStar, PathFindResultType, TileResult } from "./AStar";
 import { MiniAStar } from "./MiniAStar";
 
+const parabolaMinHeight = 50;
+
+export class ParabolaPathFinder {
+  constructor(private mg: GameMap) {}
+  private curve: DistanceBasedBezierCurve;
```

Comment:
This type is incorrect, you initialize it to undefined, so you need to declare that in the type.
```suggestion
  private curve: DistanceBasedBezierCurve | undefined;
```

I'm working on enabling safe mode, which will cause typescript to enforce this, but it's not ready yet.